### PR TITLE
feat: first implementation

### PR DESCRIPTION
**Pull request by pedronauck**
This pull request has the main package we use in all frontend packages to dev tooling and some general packages used across applications. The idea is to centralize all of them on this repo to manage the tooling better.

### CODE REVIEW

File: packages/storage/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type { Config } from '@fuels/test-lib/config';
```
Comment (luizstacio) [inline packages/storage/jest.config.ts:1]: Change the name of the package to `react-storage`

File: packages/storage/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@fuels/local-storage",
```
Comment (luizstacio) [inline packages/storage/package.json:2]: Here also `@fuels/react-storage` we can add session-storage and other types inside this package on the future.

File: packages/store/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@fuels/xstore",
```
Comment (luizstacio) [inline packages/store/package.json:2]: As this package also requires react we can change the name for `@fuels/react-xstore`

File: packages/tsconfig/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "@fuels/config-ts",
```
Comment (luizstacio) [inline packages/tsconfig/package.json:2]: `@fuels/ts-config` ??

File: packages/tsup/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "@fuels/config-tsup",
```
Comment (luizstacio) [inline packages/tsup/package.json:2]: `@fuels/tsup-config`

File: scripts/changesets/changelog.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+/**
```
Comment (luizstacio) [inline scripts/changesets/changelog.js:1]: Can we use it from package?

File: packages/test-lib/package.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "@fuels/test-lib",
```
Comment (luizstacio) [inline packages/test-lib/package.json:2]: Maybe a better name: `@fuels/jest`

Can this be used for just node projects also? If not, we can prefix it with `react`.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/store/package.json (hunk 3 above)
Comment (pedronauck) [reply packages/store/package.json:2]: ✅

File: packages/tsconfig/package.json (hunk 4 above)
Comment (pedronauck) [reply packages/tsconfig/package.json:2]: ✅

File: packages/tsup/package.json (hunk 5 above)
Comment (pedronauck) [reply packages/tsup/package.json:2]: ✅

File: packages/test-lib/package.json (hunk 7 above)
Comment (pedronauck) [reply packages/test-lib/package.json:2]: yes, it can be used for node also

File: packages/storage/jest.config.ts (hunk 1 above)
Comment (pedronauck) [reply packages/storage/jest.config.ts:1]: this isn't a react specific project, can be used anywhere

File: scripts/changesets/changelog.js (hunk 6 above)
Comment (pedronauck) [reply scripts/changesets/changelog.js]: ✅

File: packages/prettier/package.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@fuels/config-prettier",
```
Comment (luizstacio) [inline packages/prettier/package.json:2]:
```suggestion
  "name": "@fuels/prettier-config",
```

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**